### ISSUE TITLE
FCN用于混合中英文分隔的问题

### ISSUE THREAD

**Issue by binggouxsm** [WarBean/MLUtil#1]
你好，我现在遇到类似中英文混合分隔的问题，看到了你那篇论文《Chinese/English mixed Character Segmentation as Semantic Segmentation》，想尝试用FCN解决这个问题。
于是基于中文手写数据库HWDB1.1和手写数字数据集的minist，我构建了训练数据，标注好的数据如下图所示：
![8acdbda3-5d3f-42d6-9ae3-5ff5c1ce8cdd_label](https://user-images.githubusercontent.com/21152949/28577338-13b68d4c-7189-11e7-998f-28ce6c508e7a.png)

然后我基于tensorflow构建论文中的网络：

`
        keep_prob = tf.placeholder(dtype=tf.float32, shape=[], name='keep_prob')
        images = tf.placeholder(tf.float32, [None, FLAGS.image_height, FLAGS.image_width, 1],"image_batch")
        labels = tf.placeholder(tf.float32, [None, FLAGS.image_width], "label_batch")

        # 4层 down-sampling
        # 第一次池化后 12 * 128 * 16 HWC
        conv_1 = slim.conv2d(images, 16, [2, 2], 1,normalizer_fn=slim.batch_norm,normalizer_params={'is_training':True}, padding='SAME', scope='conv1')
        max_pool_1 = slim.max_pool2d(conv_1, [2, 2])
        # 6 * 64 * 32 HWC
        conv2 = slim.conv2d(max_pool_1, 32, [2, 2], 1,normalizer_fn=slim.batch_norm,normalizer_params={'is_training':True}, padding='SAME', scope='conv2')
        max_pool_2 = slim.max_pool2d(conv2, [2, 2])
        # 3 * 32 * 64 HWC
        conv3 = slim.conv2d(max_pool_2, 64, [2, 2], 1,normalizer_fn=slim.batch_norm,normalizer_params={'is_training':True}, padding='SAME', scope='conv3')
        max_pool_3 = slim.max_pool2d(conv3, [2, 2])
        # 1 * 16 * 128 HWC
        conv4 = slim.conv2d(max_pool_3, 128, [3, 2], 1,normalizer_fn=slim.batch_norm,normalizer_params={'is_training':True}, padding='SAME', scope='conv4')
        max_pool_4 = slim.max_pool2d(conv4, [3, 2])

        # 4层 up-sampling
        # 1 * 16* 128 HWC
        dconv1 = slim.conv2d_transpose(max_pool_4, 128, [1, 2], [1, 2], scope='dconv1')
        dconv2 = slim.conv2d_transpose(dconv1, 64, [1, 2], [1, 2], scope='dconv2')
        dconv3 = slim.conv2d_transpose(dconv2, 32, [1, 2], [1, 2], scope='dconv3')
        dconv4 = slim.conv2d_transpose(dconv3, 1, [1, 2], [1, 2],activation_fn=tf.nn.sigmoid,  scope='dconv4')
        # fc1 的shape 为 m * 128
        fc1 = slim.flatten(dconv4)
        # labels 的shape 为 m * 128，采用
        loss = tf.reduce_mean(tf.squared_difference(fc1, labels))

        global_step = tf.get_variable("step", [], initializer=tf.constant_initializer(0.0), trainable=False)
        rate = tf.train.exponential_decay(9e-3, global_step, decay_steps=2000, decay_rate=0.97, staircase=True)
        train_op = tf.train.AdamOptimizer(learning_rate=rate).minimize(loss, global_step=global_step)

        tf.summary.scalar('loss', loss)
        merged_summary_op = tf.summary.merge_all()
`

网络构建好之后，运用测试数据集进行训练，由于想减少训练的成本，图像选择128*24像素大小的图像。总共构建了1万个样本，训练了4000多个step，每个step 10个样本。但是 MSE cost function 从0.1降到0.01之后，就一直降不下去，明显的under-fitting。测试的效果也不好。

不知道，您是否遇到过这个问题，还是我这边网络的实现有问题。
个人怀疑是否和 切分的点（即p = 1） 相比 不切分点 （p=0）太少了，两者之间unbalanced造成的。

求指教下

**Comment by WarBean** [comment]
@binggouxsm 很抱歉现在才看到。样本unbalanced的问题我在论文里面已经说了对付方法，就是weighted cross entropy loss，你可以参考下，希望对你有用。如果还有后续问题可以给我的知乎账号（就是我的名字）私信。
